osf/models/node.py

```diff
@@ -441,12 +441,25 @@ def is_preprint(self):
         if not self.preprint_file_id or not self.is_public:
             return False
         if self.preprint_file.node_id == self.id:
-            return self.has_published_preprint
+            return self.has_published_preprint or self.has_pending_rejected_preprint
```

Comment:
It might make more sense to have `has_submitted_preprint` that checks for preprints with non-`initial` state. Since submitted preprints are a superset of published preprints, you could use that instead of both of these.